processor/deltatocumulativeprocessor/internal/delta/delta.go

```diff
@@ -0,0 +1,61 @@
+package delta
+
+import (
+	"fmt"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/data"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/streams"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+)
+
+func construct[D data.Point[D]]() streams.Aggregator[D] {
+	acc := &Accumulator[D]{dps: make(map[streams.Ident]D)}
+	lock := &Lock[D]{next: acc}
+	return lock
+}
+
+func Numbers() streams.Aggregator[data.Number] {
+	return construct[data.Number]()
+}
+
+func Histograms() streams.Aggregator[data.Histogram] {
+	return construct[data.Histogram]()
+}
+
+var _ streams.Aggregator[data.Number] = (*Accumulator[data.Number])(nil)
+
+type Accumulator[D data.Point[D]] struct {
+	dps map[streams.Ident]D
+}
+
+func (a *Accumulator[D]) Aggregate(id streams.Ident, dp D) (D, error) {
+	reset := func() {
+		a.dps[id] = dp.Clone()
+	}
+
+	aggr, ok := a.dps[id]
+	if !ok {
+		fmt.Printf("d2c: new stream %s", id)
+		reset()
+		return dp, nil
+	}
+
+	if dp.StartTimestamp() != aggr.StartTimestamp() {
+		reset()
+	}
+	if dp.Timestamp() <= aggr.Timestamp() {
+		return dp, ErrOutOfOrder{Last: aggr.Timestamp(), Sample: aggr.Timestamp()}
```

Comment:
Similarly, could we make this behavior configurable? IE, instead of erroring, it just drops the out of order metrics?

Reply:
Only the `Accumulator` call errors out with a well-known type (`ErrOutOfOrder`), which can be caught on the call-site.
I was intending for the processor to log and increase a counter, but not actually failing. How does that sound?

Reply:
Oh cool. That's perfect. I was *also* thinking it would be nice to have a set of metrics for the processor itself. Which would track errors like this. IE so a user could add alerts to know that metrics are being dropped, etc.

Reply:
yes absolutely, just put that on the roadmap

Reply:
I believe there must be a typo:
```suggestion
		return dp, ErrOutOfOrder{Last: aggr.Timestamp(), Sample: dp.Timestamp()}
```